src/senaite/jsonapi/fieldmanagers.py

```diff
@@ -571,3 +571,65 @@ def set(self, instance, value, **kw):
 
         analyses = map(api.get_object_by_uid, uids)
         self._set(instance, analyses, **kw)
+
+
+class UIDReferenceFieldManager(ATFieldManager):
+    """Adapter to get/set the value of UIDReferenceFields
+    """
+    interface.implements(IFieldManager)
+
+    def __init__(self, field):
+        super(UIDReferenceFieldManager, self).__init__(field)
+        self.multi_valued = field.multiValued
+
+    def json_data(self, instance, default={}):
+        """Get a JSON compatible value
+        """
+        value = self.get(instance)
+        if not value:
+            return default
+        if self.multi_valued:
+            out = map(api.get_url_info, value)
+        else:
+            out = value
```

Comment:
I guess `value` is in that case an UID? Or a brain? Not sure. Maybe we should return in that case a similar structure like `api.get_url_info` returns?

Reply:
The thing is that, the return value is checked in `api.to_json` function. So either we return an object or a dictionary, it will be handled without a problem. I think there is no need to repeat it.